hits_cpp/shufflesGenerator.hpp

```diff
@@ -13,24 +13,36 @@ class MemeShuffler {
             maxShuffles = len;
         }
         this->_maxShuffles = maxShuffles;
+        this->_lenPatterns=len;
+        cout << "len pattern: " << this->_lenPatterns << endl;
+        this->_numPattern=-1;
     }
 
     bool next() {
         _currentShuffle++;
+        _numPattern++;
         return hasNext();
     }
 
     Meme generate() {
         auto meme = Meme(this->_source);
         meme.getRows().clear();
-
-        auto pattern = this->_patterns[this->_currentShuffle];
+        auto pattern = this->_patterns[this->_numPattern];
         auto iter = pattern.begin();
         auto end = pattern.end();
         while (iter != end) {
             meme.getRows().push_back(this->_source.getRows()[*iter]);
             iter++;
         }
+        //check that we don't change place of the same consensus:
+        for (int i = 0; i < this->_source.getRows().size(); i++){
+            auto max_source = max_element(this->_source.getRows()[i].begin(), this->_source.getRows()[i].end()) - this->_source.getRows()[i].begin();
+            auto max_shuffle = max_element(meme.getRows()[i].begin(), meme.getRows()[i].end()) - meme.getRows()[i].begin();
+            if (max_source == max_shuffle && (this->_lenPatterns - this->_numPattern) > (this->_maxShuffles - this->_currentShuffle)){
+                _numPattern++;
```

Comment:
What happens if this is the last pattern?
What happens if no pattern matches, are we getting a random shuffles or the original PSSM?
If the original PSSM, what is the final rank of the PSSM 0 or 1.
We want the motif to be ranked as insignificant

Reply:
If all the shuffles do not do "full" shuffle so we take the last 5 shuffles of the patterns.

Reply:
Tested: https://docs.google.com/document/d/1jR-LEQTyFpKO1Yf5QYNMSiIfRVy8qB8MC07U-zhkTuw
While rank is not always 0, the client authorized the solution